crates/oxc_linter/src/rules/unicorn/prefer_set_has.rs

```diff
@@ -0,0 +1,900 @@
+        _ => false,
+    }
+}
+
+fn is_multiple_calls(node: &AstNode, ctx: &LintContext, root_scope_id: ScopeId) -> bool {
+    let mut id = node.id();
+    let mut was_in_root_scope = node.scope_id() == root_scope_id;
+    loop {
+        let Some(parent) = ctx.nodes().parent_node(id) else { return false };
+        let parent_scope = parent.scope_id();
+        if was_in_root_scope && parent_scope != root_scope_id {
+            return false;
+        }
+        was_in_root_scope = parent_scope == root_scope_id;
+        let parent_kind = parent.kind();
+        if matches!(
+            parent_kind,
+            AstKind::ForOfStatement(_)
+                | AstKind::ForInStatement(_)
+                | AstKind::ForStatement(_)
+                | AstKind::WhileStatement(_)
+                | AstKind::DoWhileStatement(_)
+                | AstKind::ArrowFunctionExpression(_)
+                | AstKind::Function(_)
+        ) {
+            return true;
+        };
+        id = parent.id();
+    }
+}
+
+impl Rule for PreferSetHas {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::VariableDeclarator(declarator) = node.kind() else {
+            return;
+        };
+
+        if let VariableDeclarationKind::Var = declarator.kind {
+            return;
+        }
+
+        let Some(init) = declarator.init.as_ref() else {
+            return;
+        };
+
+        let is_kind_of_array = is_kind_of_array_expr(init);
+
+        if !is_kind_of_array {
+            return;
+        }
+
+        let Some(ident) = declarator.id.get_binding_identifier() else {
+            return;
+        };
+
+        let Some(symbol_id) = ident.symbol_id.get() else {
+            return;
+        };
+
+        let symbol_table = ctx.symbols();
+
+        if symbol_table.get_flags(symbol_id).is_export() {
+            return;
+        }
+
+        let mut references = symbol_table.get_resolved_references(symbol_id).peekable();
+
+        let Ok(len) = references.try_len() else {
+            return;
+        };
+        if len == 0 {
+            return;
+        }
+
+        let root_scope = node.scope_id();
+
+        if len == 1 {
+            let Some(reference) = references.peek() else {
+                return;
+            };
+
+            let node = ctx.nodes().get_node(reference.node_id());
+
+            if !is_multiple_calls(node, ctx, root_scope) {
+                return;
+            }
+        }
+
+        if references.any(|reference| {
+            let node = ctx.nodes().get_node(reference.node_id());
+            let Some(parent) = ctx.nodes().parent_node(node.id()) else {
+                return true;
+            };
+            let Some(parent_of_parent) = ctx.nodes().parent_node(parent.id()) else {
+                return true;
+            };
+            let AstKind::CallExpression(call_expression) = parent_of_parent.kind() else {
+                return true;
+            };
```

Comment:
can be replaced with `parent_kind`:

```suggestion
            let Some(AstKind::CallExpression(call_expression)) = ctx.nodes().parent_kind(parent.id()) else {
                return true;
            };
```

Reply:
done